src/monitor.js

```diff
@@ -40,14 +40,45 @@ module.exports = (pluginContext) => {
   }
 
   let lastClip
-  return (env = {}) => {
-    if (isTransient()) { return Promise.resolve() }
-    const clip = getClip(env.ignoreImages)
-    if (!lastClip || lastClip.type !== clip.type || lastClip.raw !== clip.raw) {
-      lastClip = clip
-      const clipCollection = CappedClient.init(cwd, env)
-      return clipCollection.upsert(clip)
-    }
-    return Promise.resolve()
+  monitor = (env = {}) => {
+    return new Promise((resolve, reject) => {
+      if (isTransient()) {
+        resolve()
+      } else {
+        const clip = getClip(env.ignoreImages)
+        if (!lastClip || lastClip.type !== clip.type || lastClip.raw !== clip.raw) {
+          lastClip = clip
+          const clipCollection = CappedClient.init(cwd, env)
+          resolve(clipCollection.upsert(clip))
+        } else {
+          resolve()
+        }
+      }
+    })
   }
+
+  //  As there is a performance hit of clipboard on Linux platform,
+  //  Here we let Linux have more interval than other platform by default.
+  //  See also: https://github.com/tinytacoteam/zazu/issues/189
+  const DEFAULT_INTERVAL = (process.platform === 'linux' ? 3000 : 1000)
+  //  Let the minimum interval be 250ms, which is a little bit higher than
+  //  default minimum plugin system interval 100ms, for less CPU intense.
+  const MINIMUM_INTERVAL = 250
+
+  //  We use a loop here to provide user the ability to customize the interval.
+  //  it will keep looping unless got an exception.
+  start = (env = {}) => new Promise((resolve) => {
```

Comment:
Should we return this directly instead of assigning to a variable? Or `const start`?